src/main/java/com/google/devtools/build/lib/remote/RemoteStats.java

```diff
@@ -0,0 +1,49 @@
+package com.google.devtools.build.lib.remote;
+
+import com.google.devtools.build.lib.concurrent.ThreadSafety.ThreadSafe;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/** Collecting always-on stats on a remote run */
+@ThreadSafe
+public class RemoteStats {
+  AtomicInteger total_spawns;
+  AtomicInteger remote_cache_hit;
+  AtomicInteger remote_exec;
+  AtomicInteger local;
+
+  public RemoteStats() {
+    total_spawns = new AtomicInteger();
+    remote_cache_hit = new AtomicInteger();
+    remote_exec = new AtomicInteger();
+    local = new AtomicInteger();
+  }
+
+  public void addSpawn() {
+    total_spawns.incrementAndGet();
+  }
+
+  public void addRemoteCacheHit() { remote_cache_hit.incrementAndGet(); };
+
+  // Spawns remotely executed
+  public void addRemoteExec() { remote_exec.incrementAndGet(); }
+
+  // Spawns that are intentionally executed locally (as opposed to fall back mechanism)
+  public void addLocal() { local.incrementAndGet(); }
+
+  public String Summary() {
```

Comment:
Method names start with lower case.

Reply:
Is there a way to run Google Java linter on an external patch? Or just Copybara it in?

Reply:
It's automatically run on import.